pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdConnectors.java

```diff
@@ -0,0 +1,612 @@
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.net.MalformedURLException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import net.jodah.typetools.TypeResolver;
+
+@Slf4j
+@Getter
+@Parameters(commandDescription = "Interface for managing Pulsar Connectors (Ingress and egress data to and from Pulsar)")
+public class CmdConnectors extends CmdBase {
+
+    private final CreateSource createSource;
+    private final CreateSink createSink;
+    private final DeleteConnector deleteConnector;
+    private final LocalSourceRunner localSourceRunner;
+    private final LocalSinkRunner localSinkRunner;
+
+    public CmdConnectors(PulsarAdmin admin) {
+        super("connectors", admin);
+        createSource = new CreateSource();
+        createSink = new CreateSink();
+        deleteConnector = new DeleteConnector();
+        localSourceRunner = new LocalSourceRunner();
+        localSinkRunner = new LocalSinkRunner();
+
+        jcommander.addCommand("create-source", createSource);
+        jcommander.addCommand("create-sink", createSink);
+        jcommander.addCommand("delete", deleteConnector);
+        jcommander.addCommand("localrun-source", localSourceRunner);
+        jcommander.addCommand("localrun-sink", localSinkRunner);
+    }
+
+    /**
+     * Base command
+     */
+    @Getter
+    abstract class BaseCommand extends CliCommand {
+        @Override
+        void run() throws Exception {
+            processArguments();
+            runCmd();
+        }
+
+        void processArguments() throws Exception {
+        }
+
+        abstract void runCmd() throws Exception;
+    }
+
+    @Parameters(commandDescription = "Run the Pulsar Function locally (rather than deploying it to the Pulsar cluster)")
+
+    class LocalSourceRunner extends CreateSource {
+
+        @Parameter(names = "--brokerServiceUrl", description = "The URL for the Pulsar broker")
+        protected String brokerServiceUrl;
+
+        @Override
+        void runCmd() throws Exception {
+            CmdFunctions.startLocalRun(createSourceConfigProto2(sourceConfig),
+                    sourceConfig.getParallelism(), brokerServiceUrl, jarFile, admin);
+        }
+    }
+
+    class LocalSinkRunner extends CreateSink {
+
+        @Parameter(names = "--brokerServiceUrl", description = "The URL for the Pulsar broker")
+        protected String brokerServiceUrl;
+
+        @Override
+        void runCmd() throws Exception {
+            CmdFunctions.startLocalRun(createSinkConfigProto2(sinkConfig),
+                    sinkConfig.getParallelism(), brokerServiceUrl, jarFile, admin);
+        }
+    }
+
+    @Parameters(commandDescription = "Create Pulsar source connectors")
+    class CreateSource extends BaseCommand {
+        @Parameter(names = "--fqfn", description = "The Fully Qualified Function Name (FQFN) for the function")
+        protected String fqfn;
```

Comment:
isn't this leaking abstraction

Reply:
we can call it fully qualified connector name? or we can remove?  it does make it more user friendly especially fore deleting

Reply:
we should remove it